spark/src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -532,7 +532,10 @@ class DeltaAnalysis(session: SparkSession)
           newTarget,
           merge.sourceTable,
           merge.mergeCondition,
-          matchedActions ++ notMatchedActions ++ notMatchedBySourceActions
+          matchedActions ++ notMatchedActions ++ notMatchedBySourceActions,
+          // We are waiting for Spark to support the SQL "WITH SCHEMA EVOLUTION" syntax.
```

Comment:
Could we specify which version of Spark we are waiting for here? Do we have a way to remind ourselves to remember to come back here when Spark supports it? Perhaps the first step is add a "TODO(xupefei)"?

Reply:
The change is in Spark master. I don't know what will be the version number.
Added a TODO.